great_expectations/datasource/fluent/sources.py

```diff
@@ -311,7 +317,7 @@ def _read_asset_factory(
                 asset = asset_type(name=name, **kwargs)
                 self._add_asset(asset)
                 batch_request = asset.build_batch_request()
-                return self._data_context.get_validator(batch_request=batch_request)  # type: ignore[attr-defined]
+                return self._data_context.get_validator(batch_request=batch_request)  # type: ignore[arg-type] # batches.py not properly typed
```

Comment:
nit: The reason is different than the one given in great_expectations/datasource/fluent/pandas_dbfs_datasource.py. Should we make them the same?

Reply:
The other explanation is better, updated.

Reply:
Done